NashvilleHistory/www/js/controllers/MarkersCtrl.js

```diff
@@ -123,6 +119,52 @@ app.controller('MarkersCtrl', function($scope, $state, $cordovaGeolocation, Mark
       });
     }
 
+    //TODO: Get the bookmarked markers and make sure the user cannot add a marker twice to his/her bookmarks
+    function areMarkersBookmarked (){
+      BookmarkFact.getAllBookmarks(AuthFact.getUserId())
```

Comment:
rather than performing `getAllBookmarks()` on every page, why don't we lift all this work to the parent controller  inside of the `firebase.auth().onAuthStateChanged()` event listener? We could store bookmarks on the `loggedInUser` object and reference it from every child page.